File: app/src/main/java/com/jcs/where/features/message/MessageCenterActivity.java
package com.jcs.where.features.message;

import android.content.Context;
import android.net.Uri;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;

import com.blankj.utilcode.util.BarUtils;
import com.blankj.utilcode.util.ColorUtils;
import com.blankj.utilcode.util.ScreenUtils;
import com.blankj.utilcode.util.StringUtils;
import com.flyco.tablayout.SlidingTabLayout;
import com.jcs.where.R;
import com.jcs.where.base.BaseActivity;
import com.jcs.where.features.message.notice.SystemNoticeFragment;
import com.jcs.where.view.empty.EmptyView;

import io.rong.imkit.conversationlist.ConversationListFragment;

/**
 * Created by Wangsw  2021/2/19 17:07.
 * 消息中心
 */
public class MessageCenterActivity extends BaseActivity {


    private ViewPager pager;
    private SlidingTabLayout tabs_type;
    private final String[] TAB_TITLES = {StringUtils.getString(R.string.business_conversation), StringUtils.getString(R.string.system_notification)};
    private Uri mUri;



    @Override
    protected int getLayoutId() {
        return R.layout.activity_message_center;
    }

    @Override
    protected boolean isStatusDark() {
        return true;
    }

    @Override
    protected void initView() {
        BarUtils.setStatusBarColor(this, ColorUtils.getColor(R.color.white));
        pager = findViewById(R.id.pager);
        tabs_type = findViewById(R.id.tabs_type);
        pager.setOffscreenPageLimit(TAB_TITLES.length);


    }

    @Override
    protected void initData() {
        EmptyView emptyView = new EmptyView(this);
        ViewGroup.LayoutParams layoutParams = emptyView.parent_ll.getLayoutParams();
        layoutParams.width = ScreenUtils.getScreenWidth();

        InnerPagerAdapter adapter = new InnerPagerAdapter(getSupportFragmentManager(), 0);

        adapter.emptyView = emptyView;
        adapter.emptyView.showEmptyDefault();

        pager.setAdapter(adapter);
        tabs_type.setViewPager(pager, TAB_TITLES);
    }


    @Override
    protected void bindListener() {

    }


    private static class InnerPagerAdapter extends FragmentPagerAdapter {


        private EmptyView emptyView ;

        public InnerPagerAdapter(@NonNull FragmentManager fm, int behavior) {
            super(fm, behavior);
        }

        @NonNull
        @Override
        public Fragment getItem(int position) {
            if (position == 0) {
                ConversationListFragment listFragment = new ConversationListFragment();
                listFragment.setEmptyView(emptyView);

                return listFragment;
            } else {
                return new SystemNoticeFragment();
            }
        }

        @Override
        public int getCount() {
            return 2;
        }
    }


}

File: app/src/main/java/com/jiechengsheng/city/features/message/MessageCenterActivity.kt
package com.jiechengsheng.city.features.message

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.blankj.utilcode.util.ScreenUtils
import com.blankj.utilcode.util.StringUtils
import com.jiechengsheng.city.R
import com.jiechengsheng.city.base.BaseEvent
import com.jiechengsheng.city.base.EventCode
import com.jiechengsheng.city.base.mvp.BaseMvpActivity
import com.jiechengsheng.city.features.account.login.LoginActivity
import com.jiechengsheng.city.features.message.notice.SystemNoticeFragment
import com.jiechengsheng.city.storage.entity.User
import com.jiechengsheng.city.utils.Constant
import com.jiechengsheng.city.view.empty.EmptyView
import io.rong.imkit.conversationlist.ConversationListFragment
import kotlinx.android.synthetic.main.activity_message_center.*

/**
 * Created by Wangsw  2021/2/19 17:07.
 * 消息中心
 */
class MessageCenterActivity : BaseMvpActivity<MessageCenterPresenter>(), MessageCenterView {

    private val TAB_TITLES =
        arrayOf(StringUtils.getString(R.string.business_conversation), StringUtils.getString(R.string.system_notification))


    private var systemUnReadMessageCount = 0

    override fun getLayoutId(): Int {
        return R.layout.activity_message_center
    }


    override fun isStatusDark(): Boolean = true


    companion object {
        fun navigation(context: Context, systemUnreadMessage: Int) {

            val bundle = Bundle().apply {
                putInt(Constant.PARAM_COUNT, systemUnreadMessage)
            }

            val intent = if (User.isLogon()) {
                Intent(context, MessageCenterActivity::class.java)
                    .putExtras(bundle)
                    .addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
            } else {
                Intent(context, LoginActivity::class.java)
                    .addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
            }

            if (context !is Activity) {
                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
            }
            context.startActivity(intent)
        }
    }

    override fun initView() {
        systemUnReadMessageCount = intent.getIntExtra(Constant.PARAM_COUNT, 0)
        pager.offscreenPageLimit = TAB_TITLES.size

    }

    override fun initData() {
        presenter = MessageCenterPresenter(this)

        val emptyView = EmptyView(this)
        val layoutParams = emptyView.parent_ll.layoutParams
        layoutParams.width = ScreenUtils.getScreenWidth()

        val adapter = InnerPagerAdapter(supportFragmentManager, 0)
        adapter.emptyView = emptyView
        emptyView.setEmptyImage(R.mipmap.ic_empty_un_notice)
        emptyView.setEmptyHint(R.string.no_notice_yet)
        pager!!.adapter = adapter
        tabs_type!!.setViewPager(pager, TAB_TITLES)

        handleCount()

    }

    private fun handleCount() {
        if (systemUnReadMessageCount > 0) {
            is_read_iv.visibility = View.VISIBLE
        } else {
            is_read_iv.visibility = View.GONE
        }
    }

    override fun bindListener() {}

    private class InnerPagerAdapter(fm: FragmentManager, behavior: Int) : FragmentPagerAdapter(fm, behavior) {

        var emptyView: EmptyView? = null

        override fun getItem(position: Int): Fragment {
            return if (position == 0) {
                emptyView!!.showEmptyContainer()
                val listFragment = ConversationListFragment()
                listFragment.setEmptyView(emptyView)
                listFragment
            } else {
                SystemNoticeFragment()
            }
        }

        override fun getCount(): Int {
            return 2
        }
    }

    override fun onEventReceived(baseEvent: BaseEvent<*>) {
        super.onEventReceived(baseEvent)
        when (baseEvent.code) {
            EventCode.EVENT_GET_MESSAGE_COUNT -> {
                presenter.getUnreadMessageCount()
            }
            else -> {}
        }

    }

    override fun bindUnreadMessageCount(apiUnreadMessageCount: Int) {
        handleCount()
    }


}